### PR TITLE
refactor: email validations throughout application

### PR DESCRIPTION
**Pull request by bhavanakarwade**
### What ###
- Refactoring email validations throughout the application
- Applied case insensitive validations while logging onto the platform

### Why ###
- To ensure uniformity in email validation logic, making it easier to manage, update, and debug across the entire application
- To improve user experience by accommodating variations in letter case and ensuring a smoother authentication process.

### How ###
- Evaluate existing email validation implementations
- Use standardised approaches to promote code consistency and reduce redundancy

### CODE REVIEW

File: apps/api-gateway/src/authz/authz.controller.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ export class AuthzController {
   * @returns Email sent success
   */
   @Post('/verification-mail')
-  @ApiResponse({ status: 201, description: 'Success', type: ApiResponseDto })
+  @ApiResponse({ status: HttpStatus.CREATED, description: 'Success', type: ApiResponseDto })
```
Comment (vivekayanworks) [inline apps/api-gateway/src/authz/authz.controller.ts:64]: Please check if we can replace the status with HttpStatus.OK

File: apps/api-gateway/src/dtos/authDto.dto.ts [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 
 export class AuthDto {
     @ApiProperty({ example: 'awqx@getnada.com' })
-    @IsEmail()
-    @IsNotEmpty({ message: 'Please provide valid email' })
-    @IsString({ message: 'email should be string' })
+    @IsEmail({}, { message: 'Please provide a valid email' })
+    @IsNotEmpty({ message: 'Email is required' })
+    @IsString({ message: 'Email should be a string' })
+    @Transform(({ value }) =>  'string' === typeof value ? value.trim() : value)
```
Comment (nishad-ayanworks) [inline apps/api-gateway/src/dtos/authDto.dto.ts:11]: We have already created a `trim()` function in the `cast.helper.ts` file under library and applied logic of `typeof` in it. use that.

**Review by nishad-ayanworks — CHANGES_REQUESTED** [review]
Please resolve comment

File: apps/api-gateway/src/dtos/authDto.dto.ts (hunk 2 above)
Comment (bhavanakarwade) [reply apps/api-gateway/src/dtos/authDto.dto.ts]: Resolved

**Review by nishad-ayanworks — APPROVED** [review]
LG

**Pull request merged**